pkg/proc/native/proc_linux.go

```diff
@@ -177,48 +177,29 @@ func Attach(pid int, waitFor *proc.WaitFor, debugInfoDirs []string) (*proc.Targe
 	return tgt, nil
 }
 
-func isProcDir(name string) bool {
-	for _, ch := range name {
-		if ch < '0' || ch > '9' {
-			return false
-		}
-	}
-	return true
-}
-
 func waitForSearchProcess(pfx string, seen map[int]struct{}) (int, error) {
 	log := logflags.DebuggerLogger()
-	des, err := os.ReadDir("/proc")
+	pids, err := process.Pids()
```

Comment:
Why do you think this function needs to be reimplemented? Note that it is already tested (TestWaitFor in pkg/proc/proc_test.go) and works.

Reply:
In some cases, cmdline may be empty, such as a zombie process, so I want to use this library to help us handle various exceptional cases.
After reading the relevant code, I found that it doesn't need to be so complicated, I have already reverted it.

Reply:
![image](https://github.com/go-delve/delve/assets/151632624/a1142f17-464b-44f9-a9f4-b9f212e945f4)